Sources/SpeziOnboarding/OnboardingConsentView.swift

```diff
@@ -90,7 +95,7 @@ public struct OnboardingConsentView: View {
                     if !willShowShareSheet {
                         Task { @MainActor in
                             /// Stores the finished PDF in the Spezi `Standard`.
-                            await onboardingDataSource.store(exportedConsentDocumented)
+                            await onboardingDataSource.store(exportedConsentDocumented, identifier: identifier)
                             await action()
                         }
```

Comment:
I think it would make sense to do some error handling here, e.g., something like
```swift
Task {
    do {
        /// Stores the finished PDF in the Spezi `Standard`.
        try await onboardingDataSource.store(exportedConsentDocumented, identifier: identifier)
        await action()
    } catch {
        viewState = .base(.error(AnyLocalizedError(error: error)))
    }
}
```

In addition, we should make sure that the consent button is an async button that shows a progress indicator while the upload is happening and does indicate some processing state as the upload and PDF creation is working.

Reply:
Thanks again for the suggestion! I have changed it accordingly. 
Also, I implemented the second suggestion regarding the async consent button, and I believe this one is actually very crucial.  I found that pressing the consent button multiple times rapidly sometimes causes odd behavior. I could skip most of the subsequent OnboardingViews by clicking the button multiple times.

To solve this, I have now added a new ViewState ".storing", which is active while the store function has not finished yet. During .storing, the "I consent" button is now deactivated, and a ProgressView overlay is shown. If the "store" function throws an error, the button is reactivated, allowing the user to try again. I also tested this manually by throwing an Error in ExampleStandard.store, but it could be worth writing a separate test case for this.